crates/rpcsrv/src/flight/handler.rs

```diff
@@ -101,26 +106,14 @@ impl FlightSessionHandler {
         Ok(Response::new(Box::pin(stream)))
     }
 
-    pub fn new(engine: Arc<Engine>) -> Self {
-        Self {
-            engine,
-            logical_plans: DashMap::new(),
-            sessions: DashMap::new(),
-        }
-    }
-
-    async fn get_or_create_ctx<T>(
-        &self,
-        request: &Request<T>,
-    ) -> Result<Arc<Mutex<Session>>, Status> {
+    async fn get_or_create_ctx<T>(&self, request: &Request<T>) -> Result<SessionRef, Status> {
         let remote = request.remote_addr().unwrap();
 
         let ip = remote.ip().to_string();
         let port = remote.port().to_string();
         let conn_key = ConnKey { ip, port };
-
         if self.sessions.contains_key(&conn_key) {
-            let sess = self.sessions.get(&conn_key).unwrap().clone();
+            let sess = self.sessions.get(&conn_key).unwrap();
```

Comment:
This can panic. `contains_key` doesn't check if the entry is expired, but `get` will return `None` if it is.